src/app/(account)/settings/tabs/AccountSettings.tsx

```diff
@@ -1,147 +1,202 @@
-                        New Password
-                    </label>
-                    <input
-                        id="newPassword"
-                        type="password"
-                        placeholder={'New Password'}
-                        {...register('newPassword')}
-                        className="border border-gray-300 p-2"
-                    />
-                    {errors.newPassword && <span>New Password is required</span>}
-                </div>
-                <div className="flex flex-col gap-2">
-                    <label htmlFor="confirmPassword" className="font-bold">
-                        Confirm Password
-                    </label>
-                    <input
-                        id="confirmPassword"
-                        type="password"
-                        placeholder={'Confirm Password'}
-                        {...register('confirmPassword')}
-                        className="border border-gray-300 p-2"
-                    />
-                    {errors.confirmPassword && <span>Confirm Password is required</span>}
-                </div>
-                <Button type="submit" colour="orange">
-                    Update password
+        </div>
+    );
+}
+
+const passwordSchema = z
+    .object({
+        oldPassword: z.string().min(1, { message: 'Please enter your password' }),
+        newPassword: passwdSchema,
+        confirmPassword: passwdSchema,
+    })
+    .refine((data) => data.newPassword === data.confirmPassword, {
+        message: 'Passwords do not match',
+        path: ['confirmPassword'],
+    });
+function ChangePassword() {
+    const { user } = useUser();
+
+    const form = useForm<z.infer<typeof passwordSchema>>({
+        defaultValues: { oldPassword: '', newPassword: '', confirmPassword: '' },
+        resolver: zodResolver(passwordSchema),
+    });
+    const handleSubmit = form.handleSubmit((data) => {
+        user?.updatePassword({ newPassword: data.newPassword, currentPassword: data.oldPassword });
+    });
+
+    return (
+        <form className="space-y-4" onSubmit={handleSubmit}>
+            <h2 className="text-2xl font-bold">Change Password</h2>
+            <div className="mb-2 border-b-2 border-black" />
+            <ControlledField
+                control={form.control}
+                name="oldPassword"
+                label="Old Password"
+                type="password"
+            />
+            <ControlledField
+                control={form.control}
+                name="newPassword"
+                label="New Password"
+                type="password"
+            />
+            <ControlledField
+                control={form.control}
+                name="confirmPassword"
+                label="Confirm Password"
+                type="password"
+            />
+            <Button type="submit" colour="orange">
+                Update password
+            </Button>
+        </form>
+    );
+}
```

Comment:
Should display confirmation message on success, like "Successfully updated your password".